src/utils/openvpn.ts

```diff
@@ -267,26 +278,53 @@ export class VpnManager extends EventEmitter {
 		return emitter;
 	};
 
-	public start(): Bluebird<true> {
-		this.process = spawn('/usr/sbin/openvpn', this.args(), {
-			stdio: ['ignore', 'pipe', 'pipe'],
-		});
-		// proxy logs from the child process stdout/stderr
-		this.process.stdout.pipe(es.split()).on('data', data => {
-			this.emit('log', VpnLogLevels.n, data);
-		});
-		this.process.stderr.pipe(es.split()).on('data', data => {
-			this.emit('log', VpnLogLevels.n, data);
-		});
-		// proxy error events from the child process
-		this.process.on('error', err => {
-			this.emit('process:error', err);
-		});
-		this.process.on('exit', (code, signal) => {
-			this.emit('process:exit', code, signal);
+	private async killOrphan() {
+		if (!(await fs.exists(this.pidFile))) {
+			return;
+		}
+		const pid = (await fs.readFile(this.pidFile, 'utf8')).replace(/\D/g, '');
+		const signals = { PING: 0, KILL: 9 } as const;
+		const kill = (signal?: ValueOf<typeof signals>) =>
+			spawn('/bin/kill', signal != null ? [`-${signal}`, pid] : [pid]);
+
+		return new Promise(resolve => {
+			const timeout = Date.now() + 5 * 1000;
+			const waitForDeath = (code?: number) => {
+				if (code !== 0) {
```

Comment:
If it doesn't equal 0?

Reply:
Yeah, if the signal was _not_ successfully sent (i.e., the process does not exist, it has been killed and our work here is done).

Reply:
Ok, can you add that as a comment please.  Also it doesn't cover cases where the issue is in `kill` itself, but that should be pretty rare and we'll just fail later and restart the work/retry the kill

Reply:
@Page- comment added, and I've also handled the (theoretical) case where a process refuses to term/kill in a timely manner.